addons/website_membership/controllers/main.py

```diff
@@ -77,47 +72,43 @@ def members(self, membership_id=None, country_name=None, country_id=0, page=0, *
         })
 
         # format domain for group_by and memberships
-        membership_ids = product_obj.search(cr, uid, [('membership', '=', True)], order="website_sequence", context=context)
-        memberships = product_obj.browse(cr, uid, membership_ids, context=context)
+        membership_ids = Product.search([('membership', '=', True)], order="website_sequence")
```

Comment:
use membership instead of membership_ids variable, it's return recordsets not list of ids.